Delete outdated files

kasohrab opened the pull request:
## Description
Title. For fun and historical reasons, I'll leave comments guessing why we previously used these files.

## Associated / Resolved Issue
Resolves [ClickUp card](https://app.clickup.com/t/2jmad9y)

## Steps to test
### Test Case 1
1. make clean && make perf
2. run sim
3. no regressions

## Key Files to Review
basically just check if we need anyone of them still (I doubt it) or if there's anything else I should delete.
## Review Checklist

- [x] **Docstrings**: All methods and classes should have the file appropriate docstrings which follow the guidelines in the ["Contributing" page](https://rj-rc-software.readthedocs.io/en/latest/contributing.html) of our docs.
- [x] **Remove extra print statements**: Any print statements used for debugging should be removed
- [x] **Tag reviewers**: Tag some people for review and ping them on Slack

**Expected result:** works

kasohrab committed:
delete outdated files

kasohrab commented on autoupdate-docs.sh:
```diff
@@ -1,15 +0,0 @@
-git config --global user.email $GIT_EMAIL
-mkdir -p api_docs
-git clone -b gh-pages git://github.com/robojackets/robocup-software api_docs/html
-make apidocs
```
we used to host docs on the gh-pages branch

kasohrab commented on cmake/MbedTargets.cmake:
```diff
@@ -1,1304 +0,0 @@
-#
-# ***** DO NOT MANUALLY EDIT THIS FILE *****
```
robot previously used mbed as MCU.

kasohrab commented on cmake/MbedUtil.cmake:
```diff
@@ -1,284 +0,0 @@
-# Set the mbed platform we should compile for.
-# Usage:
-#   mbed_set_platform(<var-name-prefixt> TARGET <target-id>)
-function(MBED_SET_PLATFORM target_var_base)
```
firmware and software used to be a monorepo

kasohrab commented on robocup.sublime-project:
```diff
@@ -1,63 +0,0 @@
-{
```
people liked sublime

kasohrab commented on runcomp.sh:
```diff
@@ -1,29 +0,0 @@
-#!/bin/bash
-
-if ! command -v git >/dev/null; then
```
from before ros2, custom launch script

kasohrab commented on util/arch-setup:
```diff
@@ -1,41 +0,0 @@
-
-BASE=$(readlink -f $(dirname $0)/..)
-
-echo "-- Installing grSim default configuration"
```
people liked arch

kasohrab commented on util/osx-setup:
```diff
@@ -1,52 +0,0 @@
-
-BASE="$(dirname $0)/.."
-
-if [ ! $(which brew) ]; then
```
before ros2 we could support osx.

kasohrab commented on util/profile:
```diff
@@ -1,43 +0,0 @@
-
-echo "Running $*"
-
-# We have this mess because there is no way to get the PID
```
no idea lol

kasohrab commented on cmake/FindBullet.cmake:
```diff
@@ -1,77 +0,0 @@
-# - Try to find the Bullet physics engine
```
we used to have our own simulator, then we switched to grSim (now we use erforce's sim). Our 2012 and 2013 [tdps](https://wiki.robojackets.org/Team_Description_Papers) talk about it a good bit

p-nayak11 commented on soccer/CMakeLists.txt:
```diff
@@ -37,9 +37,6 @@ find_package(PythonLibs 3.2 REQUIRED)
 find_package(Eigen3 REQUIRED)
 
-# libusb
-find_package(libusb-1.0 REQUIRED)
```
I assume that we don't need to communicate with a USB anymore?

p-nayak11 reviewed (changes requested):
All of the deletions make sense. I believe the rj_gameplay/tests folder is something that should be considered for scrapping as well since we don't use it at all (unless it has a purpose?).

kasohrab committed:
more cleanup

kasohrab committed:
Delete README.md

at some point we should design tests for gameplay, but right now they are not really necessary. Most of our code QA can be done with the testing tab.

kasohrab commented:
> All of the deletions make sense. I believe the rj_gameplay/tests folder is something that should be considered for scrapping as well since we don't use it at all (unless it has a purpose?).

I deleted all the tests that are outdated, the test-rc.py is still useful. I'm not sure if our CI actually runs that test right now 😅, but still it's a good regression test to have. We can always design a good testing framework later, so I see no issue with deleting the outdated stuff right now.

kasohrab commented on soccer/CMakeLists.txt:
```diff
@@ -56,39 +53,6 @@ find_package(
 
-# find boost python This package is named 'python3' on Arch, 'python-py34' on Ubuntu 14, and
-# 'python' on OS X
-set(FOUND_A_BOOST_PYTHON FALSE)
```
we use ros2 now, and no longer need boost python

kasohrab committed:
I suppose we can delete this too

kasohrab committed:
Merge branch 'summer-cleaning' of https://github.com/RoboJackets/robocup-software into summer-cleaning

p-nayak11 reviewed (approved):
LGTM! I think referring this PR in the docs log or somewhere else would be a good idea to keep track of how the codebase is changing while we are at GT.

kfu02 reviewed (approved):
can't believe we used to have firmware sounds

kasohrab committed:
we don't need to update udev rules anymore
